### PR TITLE
update detekt to RC12

### PR DESCRIPTION
**Pull request by arturbosch**
Allows to remove suppression in JvmSynchronized.

### CODE REVIEW

**Comment by robstoll** [comment]
Thanks for the PR. Is it still possible to have an overview over all subprojects with RC12?
If yes, then please add one to `"$rootProject.buildDir/detekt/detekt-checkstyle.xml" 
If not, then sonar.kotlin.detekt.reportPaths in .travis.yml needs to be adopted.

**Comment by robstoll** [comment]
If you don't mind you could remove more TODOs (to see if they are really fixed in RC12). Search for `TODO remove`

**Comment by arturbosch** [comment]
Oh, there are more TODO's :).
Seems like most of them expect two can be removed.
For one I already have a fix upstream in detekt.
The other one wll get fixed next. 
I will update this PR when RC13 is out next week :).

File: gradle/scripts/detekt.yml [hunk 1]
```diff
@@ -35,6 +34,10 @@ potential-bugs:
     UnsafeCast:
         active: false
 
+performance:
+    ArrayPrimitive:
+        active: false
```
Comment (arturbosch) [inline gradle/scripts/detekt.yml]: There were three new kinds of issues in your code (mainly in test cases), which you have to decide what to do about them ^^

**Comment by arturbosch** [comment]
Hey, update to RC14 is ready with all TODO removed :)

**Comment by robstoll** [comment]
Thanks for the PR :)

**Pull request merged**